internal/datastore/mongo_store.go

```diff
@@ -114,14 +144,86 @@ func (mongodb MongoDB) DeleteQueue(id db.QueueId) error {
 	return nil
 }
 
-func (mongodb MongoDB) AddTokenToQueue(db.QueueId, db.Token) {
-	panic("Not implemented")
+func (mongodb MongoDB) AddTokenToQueue(id db.QueueId, token db.Token) (db.TokenId, error) {
+	token.Id = ""
+	token.QueueId = id
+
+	// Lock queue to ensure 2 concurrent add tokens don't happen
+	queueMutex.Lock()
+
+	// if there were 2 concurrent calls to this function, 2 tokens might get the same number
+	max, err := mongodb.GetMaxToken(id)
+
+	if err != nil {
+		log.Fatal(err)
+		queueMutex.Unlock()
+		return token.Id, err
+	}
+
+	token.TokenNumber = max + 1
+
+	result, err := mongodb.Token.InsertOne(context.TODO(), token)
+
+	queueMutex.Unlock()
+
+	if err != nil {
+		return token.Id, err
+	}
```

Comment:
Let's log a description of the error here so that we're aware that the next step was missed and the reason.